### DISCUSSION TITLE
Rust gamedev patterns

### DISCUSSION THREAD

**Discussion by erlend-sh** [rust-gamedev/rust-game-ports#128] [Ideas]
From the readme:
> While some ports are straightforward, for example Rust+immediate mode library ("IML") -> Rust+IML, other types of port can be subtly challenging:

An natural side effect of putting a lot of different rust games in one place is that their similarities and differences become more evident.

@64kramsystem since it sounds like you’ve already mapped a lot of this out in your mind, could we get a Patterns or Architectures section added to the readme? It would just contain sub-headers for all the toolkits featured here, plus a paragraph or two explaining the essence of their approach to game coding.

You’ve already alluded to these different patterns, but so far we’ve refrained from clearly spelling out which particular patterns each individual toolkit adheres to.

**Comment by 64kramsystem** [comment]
I actually have a guide to the Rust game engines in the pipeline, which I think would fit this description.

I can't get into details because the work gets too much (and it wouldn't be different from the examples already provided by the libraries), but a high level, comparative, description, is definitely something that I will do.

I will take a bit because I need to test Tetra, to verify presumed Winit issues with fullscreen handling.
